director/director.go

```diff
@@ -69,7 +72,70 @@ func (d *Director) runDistributeListener() {
 }
 
 func (d *Director) distributeChecks() error {
-	log.Infof("%v-distributeChecks: Distributing checks across all members in cluster", d.Identifier)
+	log.Debugf("%v-distributeChecks: Performing member existence verification", d.Identifier)
+
+	if err := d.verifyMemberExistence(); err != nil {
+		return fmt.Errorf("%v-distributeChecks: Unable to verify member existence in cluster: %v",
+			d.Identifier, err.Error())
+	}
+
+	log.Infof("%v-distributeChecks: Performing check distribution across members in cluster", d.Identifier)
+
+	// fetch all members in cluster
+	members, err := d.DalClient.GetClusterMembers()
+	if err != nil {
+		return fmt.Errorf("Unable to fetch cluster members: %v", err.Error())
+	}
+
+	if len(members) == 0 {
+		return fmt.Errorf("No active cluster members found - bug?")
+	}
+
+	log.Debugf("%v-distributeChecks: Distributing checks between %v cluster members", d.Identifier, len(members))
+
+	// fetch all check keys
+	checkKeys, err := d.DalClient.GetCheckKeys()
+	if err != nil {
+		return fmt.Errorf("Unable to fetch all check keys: %v", err.Error())
+	}
+
+	if len(checkKeys) == 0 {
+		return fmt.Errorf("Check configuration is empty - nothing to distribute!")
+	}
+
+	if err := d.performCheckDistribution(members, checkKeys); err != nil {
+		return fmt.Errorf("Unable to complete check distribution: %v", err.Error())
+	}
+
+	return nil
+}
+
+// A simple (and equal) check distributor
+//
+// Divide checks equally between all members; last member gets remainder of checks
+func (d *Director) performCheckDistribution(members, checkKeys []string) error {
```

Comment:
Ahh - this is the current check distribution logic:

- Take all available checks $all_available_checks
- Divide those by $num_members, to figure out how many checks each member will get
- Loop through each member, assign it the pre-determined number of checks
- Last member gets the remainder of all checks

This still needs to be finished up to actually set the details in etcd, but the main logic bit is complete.